internal/dataaccess/expense_dao.go

```diff
@@ -0,0 +1,161 @@
+package dataaccess
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Yokithaii/CLI_Expense_Tracker/internal/entities"
+	"github.com/jackc/pgx/v5"
+)
+
+type ExpenseDAO struct {
+	conn *pgx.Conn
+}
+
+func (e *ExpenseDAO) CloseConnect(context context.Context) {
+	e.conn.Close(context)
+}
+
+// Returns a connection to local DB.
+func NewConnectionToDB() (*ExpenseDAO, error) {
+	conn, err := pgx.Connect(context.Background(), "postgres://postgres:Asdewq232113@localhost:5433/postgres")
+
+	if err != nil {
+		return nil, err
+	}
+	return &ExpenseDAO{conn: conn}, err
+}
+
+// Checks that expense if provided id exists.
+func (e *ExpenseDAO) ExpenseExists(id int) (bool, error) {
```

Comment:
```suggestion
func (e *ExpenseDAO) ExpenseExists(ctx context.Context, id int) (bool, error) {
```

В `CloseConnect` ты передал контекст параметром. Ну тогда уж следуй такой схеме в остальных методах и передавай контекст параметром.

Тут примерно такая же история как с connection string - приложение решает, какой контекст использовать. А dataacess работает с тем контекстом, который ему спустили свыше.